common/network-types/src/session/state.rs

```diff
@@ -209,7 +202,7 @@ pub struct SessionState<const C: usize> {
     frame_reassembler: Arc<FrameReassembler>,
     cfg: SessionConfig,
     #[pin]
-    segment_ingress: UnboundedSender<SessionMessage<C>>,
+    segment_egress_send: UnboundedSender<SessionMessage<C>>,
```

Comment:
Unfortuntately, in our case we need to use `unbounded_send` which is not async, in order for the `retransmit_segments` to not be async as well, because it is being called from the non-async `start_send` here:

https://github.com/hoprnet/hoprnet/blob/43612b6860eda3d7067467e4a95f6020bbb6400c/common/network-types/src/session/state.rs#L567